lib/block-supports/typography.php

```diff
@@ -451,51 +458,85 @@ function gutenberg_get_typography_font_size_value( $preset, $should_use_fluid_ty
 	$default_minimum_font_size_factor = 0.75;
 	$default_maximum_font_size_factor = 1.5;
 	$default_scale_factor             = 1;
+	$default_minimum_font_size_limit  = '14px';
 
 	// Font sizes.
-	$fluid_font_size_settings = isset( $preset['fluid'] ) ? $preset['fluid'] : null;
+	$fluid_font_size_settings = isset( $font_size['fluid'] ) ? $font_size['fluid'] : null;
 
 	// A font size has explicitly bypassed fluid calculations.
 	if ( false === $fluid_font_size_settings ) {
-		return $preset['size'];
+		return $font_size['size'];
 	}
 
 	// Try to grab explicit min and max fluid font sizes.
 	$minimum_font_size_raw = isset( $fluid_font_size_settings['min'] ) ? $fluid_font_size_settings['min'] : null;
 	$maximum_font_size_raw = isset( $fluid_font_size_settings['max'] ) ? $fluid_font_size_settings['max'] : null;
 
-	// Font sizes.
-	$preferred_size = gutenberg_get_typography_value_and_unit( $preset['size'] );
+	// Parses default font size.
+	$preferred_size = gutenberg_get_typography_value_and_unit( $font_size['size'] );
 
 	// Protect against unsupported units.
 	if ( empty( $preferred_size['unit'] ) ) {
-		return $preset['size'];
+		return $font_size['size'];
 	}
 
-	// If no fluid min or max font sizes are available, create some using min/max font size factors.
+	// If no fluid min font size is available, create one using min font size factor.
 	if ( ! $minimum_font_size_raw ) {
-		$minimum_font_size_raw = ( $preferred_size['value'] * $default_minimum_font_size_factor ) . $preferred_size['unit'];
+		$minimum_font_size_raw = round( $preferred_size['value'] * $default_minimum_font_size_factor, 3 ) . $preferred_size['unit'];
 	}
 
+	// Parses the minimum font size limit, so we can perform checks using it.
+	$minimum_font_size_limit = gutenberg_get_typography_value_and_unit(
+		$default_minimum_font_size_limit,
+		array(
+			'coerce_to' => $preferred_size['unit'],
+		)
+	);
+
+	/*
+	 * Checks if a user-defined font size is lower than $minimum_font_size_limit. If so than it should become the minimum font size.
```

Comment:
Nit: tiny typo, also might be worth splitting onto a new line starting from `If so` since this line is pretty long?

```suggestion
	 * Checks if a user-defined font size is lower than $minimum_font_size_limit. If so then it should become the minimum font size.
```